draco/synthesis/noise.py

```diff
@@ -61,6 +67,59 @@ def process(self, data):
         return data
 
 
+class GaussianNoiseDataset(task.SingleTask):
+    """Generates a Gaussian distributed noise dataset using the
+    the noise estimates of an existing dataset.
+
+    Attributes
+    ----------
+    seed : int
+        Random seed for the noise generation.
+    """
+
+    seed = config.Property(proptype=int, default=None)
+
+    def process(self, data):
+        """Generates a Gaussian distributed noise dataset,
+        given the provided dataset's visibility weights
+
+        Parameters
+        ----------
+        data : :class:`containers`
+            Any dataset which contains a vis and weight attribute.
+            Note the modification is done in place.
+
+        Returns
+        -------
+        data_noise : same as :param:`data`
+            The previous dataset with the visibility replaced with
+            a Gaussian distributed noise realisation.
+
+        """
+        # Distribute in something other than `stack`
+        data.redistribute("freq")
```

Comment:
Because now we need to loop over stacks to find the auto-correlations, I thought it simpler to ensure we are not distributed in `stack` so I put this `redistribute` call back here.